---
title: "monarchr"
author: "charles carey"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

[![Build Status](https://travis-ci.org/charlieccarey/monarchr.svg?branch=master)](https://travis-ci.org/charlieccarey/monarchr) [![codecov](https://codecov.io/github/charlieccarey/monarchr/branch/master/graphs/badge.svg)](https://codecov.io/github/charlieccarey/monarchr) 

This package is in early development.

## Introduction.

This is the README for the monarchr package.

We use monarchr to retrieve and work with genetic, phenotype, and ontological 
information from the Monarch Inititiative [monarchinitiative.org](https://monarchinitiative.org).

Monarch Initiative's API provides structured data from many biological databases. 
Some of the key advantages of Monarch ensue from their considerable effort to enable 
cross species lookups and unification of many different, often species specific ontologies.

The monarchr package has the following goals:

1. Query Monarch's web API, returning R objects for further analysis in R.
2. Ease access from other popular bioconductor packages to Monarch's data.
3. Facilitate reuse of Monarch's data by other packages.

[view documentation](https://charlieccarey.github.io/monarchr/index.html)

[view code](https://github.com/charlieccarey/monarchr)

[Also see the swagger codegen autogenerated client to the api](https://github.com/charlieccarey/monarchr.biolink)

## Installation

```{r eval=FALSE}
install.packages("devtools")
devtools::install_github("charlieccarey/monarchr")
```

## Example: Get homologs to a particular gene.

Often, we are interested in a particular gene's homologs or orthologs in another organisms.

We query Monarch by using the bioentity_homologs function.

```{r ex_homologs}
library(monarchr)
gene <- "NCBIGene:8314"
results <- bioentity_homologs(gene)
```

The results are a list comprised of:

- A tibble of homologs (paralogs, orthologs and homologs).
- A response object (for troubleshooting and to access the raw response headers and json content if necessary)

When we ran this query, we retrieved ```r nrow(results$homologs)``` homologs.

```{r ex_homologs_resu, echo = TRUE}
library(DT)
datatable(results$homologs)
```

## More info and Troubleshooting.

See troubleshooting and other articles in documentation.

The troubleshooting article demonstrates how to verify that our homologs match 
the results given at Monarch Initiative's website.
