---
title: '`monarchr` - An R package for Monarch'
author: "Charlie Carey and Ted Laderas"
date: "4/13/2018"
output:
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

R/[Bioconductor](https://www.bioconductor.org) is highly used in bioinformatics workflows, allowing users to quickly connect identifiers that are technology specific to gene/transcript/protein identifiers. However, mapping disease phenotypes is a relatively underserved area in the Bioconductor Annotation Framework. An R-based interface to the Human Phenotype Ontology (HPO) would not only increase use, but also enable better open science. 

We are suggesting to develop the `monarchr` based on the [Bioconductor annotation workflow](https://www.bioconductor.org/help/workflows/annotation-data/).

## Starting Steps: Using Swagger on the Biolink API

[Swagger Codegen](https://swagger.io/swagger-codegen/) is a code generation framework that is helpful in developing API clients for any API that fits the OpenAPI specification. API clients can be developed, tested and deployed in an automated fashion for a number of languages (including R, Python and Rust). 

Using Swagger Codegen on the Biolink API will be the starting point for the `monarchr` package, as it generates stubs and auto-documentation in the R Package format. However, putting together the functions into use cases that are useful 

## Development of Monarch Use Cases

Swagger Codegen is just the starting point for `monarchr`. The real meat of development will be directed by use cases to utilize the HPO. This development will be `vignette`/documentation driven, where Monarch developers will describe a use case for `monarchr` (such as mapping disease homologs).

## Submission of `monarchr` as a Bioconductor annotation package

We are recommending that the package be part of [Bioconductor](https://www.bioconductor.org) as it has the largest potential audience of users who would want to leverage the HPO. Doing this would add needed resource/development time to `monarchr`. 

[Submission Guidelines to Bioconductor](https://www.bioconductor.org/developers/package-guidelines/) are fairly strict. The data structures output should be current Bioconductor objects (such as `ExpressionSet` and `AnnotatedDataFrame`). Bioconductor also has strict instructions on [web query packages](https://bioconductor.org/developers/how-to/web-query/) to ensure that the package continues to work. 

Packages must also pass their nightly build tests on a number of machines. Using a continuous integration framework such as [Travis CI](https://travis-ci.org/getting_started) will help alleviate this.  

# Future Directions: Python Client 

Depending on funding, it is possible to use Swagger to generate an API Client for Python as well.
