---
title: "Simple Example: info for a gene"
author: "Charles Carey"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Simple Example: info for a gene}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# GOAL: show homologs, diseases, phenotypes, anatomy, pathways and interactions associated with a gene.
We demonstrate these associations which correspond to tabs from Monarch Initiative
website that result from a search at the website for the human gene NCBIGene:8314.

## Loading monarchr client library and libraries for displaying tables.

(DT is the datatables package which we use to display the results.
knitr includes kable, another way to generate tables.)

```{r} 
library(monarchr)
library(DT)
library(knitr)
gene <- "NCBIGene:8314"
```

## Get the homologs (paralogs, orthologs and homologs) to a gene.

Homologs of gene, displayed using DataTable.

```{r homologs}
homs <- bioentity_homologs(gene)
datatable(homs$homologs)
```

## Get the diseases associated with a gene.

Diseases associated with gene, displayed using kable.

Note, the publications look clickable, but are not. This is a bug in the formatting. They are meant to be raw text.

```{r diseases}
dis <- bioentity_diseases_assoc_w_gene(gene)
kable(dis$diseases)
```

## Get the phenotypes associated with a gene.

Phenotypes associated with gene, displayed using kable.

Note, the publications look clickable, but are not. This is a bug in the formatting. They are meant to be raw text.

```{r phenotypes}
phe <- bioentity_phenotypes_assoc_w_gene(gene)
kable(head(phe$phenotypes))
```


## Get the anatomy associated with a gene.

Anatomy associated with gene, displayed using kable.

Molecularly defined by expression data?

```{r anatomy}
anat <- bioentity_exp_anatomy_assoc_w_gene(gene)
kable(head(anat$anatomy))
```

## Get the pathways associated with a gene.

Pathways associated with gene, displayed using kable.

```{r pathways}
pth <- bioentity_pathways_assoc_w_gene(gene)
kable(pth$pathways)
```

## Get the interactions associated with a gene.

Interactions associated with gene, displayed using kable.

Note, the publications look clickable, but are not. This is a bug in the formatting. They are meant to be raw text.

```{r interactions}
intx <- bioentity_interactions_assoc_w_gene(gene)
kable(head(intx$interactions))
```
