<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interface to Monarch Initiative API for Phenotype Ontology and Genotype • monarchr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><meta property="og:title" content="Interface to Monarch Initiative API for Phenotype Ontology and Genotype">
<meta property="og:description" content="The Monarch Initiative integrates and manages ontologies,
    phenotypes, genetics and genotypes across many species from many 
    resources. The monarchr package uses the monarch web API to retrieve 
    that information into an R environment.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">monarchr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/disease_to_model_organism.html">disease_to_model_organism</a>
    </li>
    <li>
      <a href="articles/proposal.html">proposal: `monarchr` - An R package for Monarch</a>
    </li>
    <li>
      <a href="articles/simple_ex_gene_info.html">Simple Example: info for a gene</a>
    </li>
    <li>
      <a href="articles/troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/charlieccarey/monarchr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="index_files/htmlwidgets-1.2/htmlwidgets.js"></script><script src="index_files/jquery-1.12.4/jquery.min.js"></script><link href="index_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="index_files/datatables-binding-0.4/datatables.js"></script><link href="index_files/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="index_files/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="index_files/dt-core-1.10.16/js/jquery.dataTables.min.js"></script><link href="index_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="index_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    

    
    
<p><a href="https://travis-ci.org/charlieccarey/monarchr"><img src="https://travis-ci.org/charlieccarey/monarchr.svg?branch=master" alt="Build Status"></a> <a href="https://codecov.io/github/charlieccarey/monarchr"><img src="https://codecov.io/github/charlieccarey/monarchr/branch/master/graphs/badge.svg" alt="codecov"></a></p>
<p>This package is in early development.</p>
<div id="introduction-" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-" class="anchor"></a>Introduction.</h2>
<p>This is the README for the monarchr package.</p>
<p>We use monarchr to retrieve and work with genetic, phenotype, and ontological information from the Monarch Inititiative <a href="https://monarchinitiative.org">monarchinitiative.org</a>.</p>
<p>Monarch Initiative’s API provides structured data from many biological databases. Some of the key advantages of Monarch ensue from their considerable effort to enable cross species lookups and unification of many different, often species specific ontologies.</p>
<p>The monarchr package has the following goals:</p>
<ol style="list-style-type: decimal">
<li>Query Monarch’s web API, returning R objects for further analysis in R.</li>
<li>Ease access from other popular bioconductor packages to Monarch’s data.</li>
<li>Facilitate reuse of Monarch’s data by other packages.</li>
</ol>
<p><a href="https://charlieccarey.github.io/monarchr/index.html">view documentation</a></p>
<p><a href="https://github.com/charlieccarey/monarchr">view code</a></p>
<p><a href="https://github.com/charlieccarey/monarchr.biolink">Also see the swagger codegen autogenerated client to the api</a></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">"devtools"</span>)
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"charlieccarey/monarchr"</span>)</code></pre></div>
</div>
<div id="example-get-homologs-to-a-particular-gene-" class="section level2">
<h2 class="hasAnchor">
<a href="#example-get-homologs-to-a-particular-gene-" class="anchor"></a>Example: Get homologs to a particular gene.</h2>
<p>Often, we are interested in a particular gene’s homologs or orthologs in another organisms.</p>
<p>We query Monarch by using the bioentity_homologs function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(monarchr)
gene &lt;-<span class="st"> "NCBIGene:8314"</span>
results &lt;-<span class="st"> </span><span class="kw"><a href="reference/bioentity_homologs.html">bioentity_homologs</a></span>(gene)</code></pre></div>
<p>The results are a list comprised of:</p>
<ul>
<li>A tibble of homologs (paralogs, orthologs and homologs).</li>
<li>A response object (for troubleshooting and to access the raw response headers and json content if necessary)</li>
</ul>
<p>When we ran this query, we retrieved <code>17</code> homologs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DT)
<span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(results<span class="op">$</span>homologs)</code></pre></div>
<div id="htmlwidget-39fbeaac3307c573818a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-39fbeaac3307c573818a">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17"],["Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens"],["HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950"],["BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1"],["in paralogy relationship with","in paralogy relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in paralogy relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with"],["UCHL5","UCHL3","BAP1","Bap1","UCH2","Bap1","BAP1","UCHL1","bap1","BAP1","BAP1","BAP1","BAP1","UCH1","bap1","bap1","calypso"],["HGNC:19678","HGNC:12515","NCBIGene:100051934","RGD:1311938","NCBIGene:842876","MGI:1206586","NCBIGene:100124510","HGNC:12513","Xenbase:XB-GENE-494978","NCBIGene:460426","NCBIGene:100030960","NCBIGene:100154798","NCBIGene:697221","NCBIGene:831492","NCBIGene:100557123","ZFIN:ZDB-GENE-050208-492","FlyBase:FBgn0262166"],["Homo sapiens","Homo sapiens","Equus caballus","Rattus norvegicus","Arabidopsis thaliana","Mus musculus","Bos taurus","Homo sapiens","Xenopus (Silurana) tropicalis","Pan troglodytes","Monodelphis domestica","Sus scrofa","Macaca mulatta","Arabidopsis thaliana","Anolis carolinensis","Danio rerio","Drosophila melanogaster"],["phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","sequence similarity evidence, phylogenetic evidence, protein BLAST evidence used in manual assertion","phylogenetic evidence"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"ZFIN:ZDB-PUB-030905-1",null],["panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther, zfin","panther"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>subject.taxon<\/th>\n      <th>subject.id<\/th>\n      <th>subject<\/th>\n      <th>relation<\/th>\n      <th>object<\/th>\n      <th>object.id<\/th>\n      <th>object.taxon<\/th>\n      <th>evidence<\/th>\n      <th>publications<\/th>\n      <th>provided_by<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="more-info-and-troubleshooting-" class="section level2">
<h2 class="hasAnchor">
<a href="#more-info-and-troubleshooting-" class="anchor"></a>More info and Troubleshooting.</h2>
<p>See troubleshooting and other articles in documentation.</p>
<p>The troubleshooting article demonstrates how to verify that our homologs match the results given at Monarch Initiative’s website.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/charlieccarey/monarchr">https://​github.com/​charlieccarey/​monarchr</a>
</li>
<li>Report a bug at <br><a href="https://github.com/charlieccarey/monarchr/issues">https://​github.com/​charlieccarey/​monarchr/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>CC0</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Charles C Carey <br><small class="roles"> Maintainer, author </small>  </li>
<li>Ted Laderas <br><small class="roles"> Author </small>  </li>
</ul>
</div>

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Charles C Carey, Ted Laderas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
