<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interface to Monarch Initiative API for Phenotype Ontology and Genotype • monarchr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><meta property="og:title" content="Interface to Monarch Initiative API for Phenotype Ontology and Genotype">
<meta property="og:description" content="The Monarch Initiative integrates and manages ontologies,
    phenotypes, genetics and genotypes across many species from many 
    resources. The monarchr package uses the monarch web API to retrieve 
    that information into an R environment.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">monarchr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/monarchr-simple-ex.html">Simple Example</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="index_files/htmlwidgets-1.0/htmlwidgets.js"></script><script src="index_files/jquery-1.12.4/jquery.min.js"></script><link href="index_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="index_files/datatables-binding-0.4/datatables.js"></script><link href="index_files/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="index_files/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="index_files/dt-core-1.10.16/js/jquery.dataTables.min.js"></script><link href="index_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="index_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    

    
    
<div id="introduction-" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-" class="anchor"></a>Introduction.</h2>
<p>This is the README for the monarchr package.</p>
<p>We use monarchr to retrieve and work with genetic, phenotype, and ontological information from the Monarch Inititiative <a href="https://monarchinitiative.org">monarchinitiative.org</a>.</p>
<p>Monarch Initiative’s API provides structured data from many biological databases. Some of the key advantages of Monarch ensue from their considerable effort to enable cross species lookups and unification of many different, often species specific ontologies.</p>
<p>The monarchr package has the following goals:</p>
<ol style="list-style-type: decimal">
<li>Query Monarch’s web API, returning R objects for further analysis in R.</li>
<li>Ease access from other popular bioconductor packages to Monarch’s data.</li>
<li>Facilitate reuse of Monarch’s data by other packages.</li>
</ol>
</div>
<div id="examples-" class="section level2">
<h2 class="hasAnchor">
<a href="#examples-" class="anchor"></a>Examples.</h2>
<p>Here are a few examples of usage.</p>
<div id="get-homologs-to-a-particular-gene-" class="section level3">
<h3 class="hasAnchor">
<a href="#get-homologs-to-a-particular-gene-" class="anchor"></a>Get homologs to a particular gene.</h3>
<p>Often, we are interested in a particular gene’s homologs or orthologs in another organisms.</p>
<p>We query Monarch by using the bioentity_homologs function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(monarchr)
gene &lt;-<span class="st"> "NCBIGene:8314"</span>
results &lt;-<span class="st"> </span><span class="kw"><a href="reference/bioentity_homologs.html">bioentity_homologs</a></span>(gene)</code></pre></div>
<p>The results are a tibble of homologs (paralogs, orthologs and homologs).</p>
<p>When we ran this query, we retrieved <code>17</code> homologs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DT)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(results<span class="op">$</span>homologs)</code></pre></div>
<div id="htmlwidget-ae081bea80457c21ba5f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ae081bea80457c21ba5f">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17"],["Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens"],["HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950"],["BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1"],["in paralogy relationship with","in paralogy relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in paralogy relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with"],["UCHL5","UCHL3","BAP1","Bap1","UCH2","Bap1","BAP1","UCHL1","bap1","BAP1","BAP1","BAP1","BAP1","UCH1","bap1","bap1","calypso"],["HGNC:19678","HGNC:12515","NCBIGene:100051934","RGD:1311938","NCBIGene:842876","MGI:1206586","NCBIGene:100124510","HGNC:12513","Xenbase:XB-GENE-494978","NCBIGene:460426","NCBIGene:100030960","NCBIGene:100154798","NCBIGene:697221","NCBIGene:831492","NCBIGene:100557123","ZFIN:ZDB-GENE-050208-492","FlyBase:FBgn0262166"],["Homo sapiens","Homo sapiens","Equus caballus","Rattus norvegicus","Arabidopsis thaliana","Mus musculus","Bos taurus","Homo sapiens","Xenopus (Silurana) tropicalis","Pan troglodytes","Monodelphis domestica","Sus scrofa","Macaca mulatta","Arabidopsis thaliana","Anolis carolinensis","Danio rerio","Drosophila melanogaster"],["phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","sequence similarity evidence, phylogenetic evidence, protein BLAST evidence used in manual assertion","phylogenetic evidence"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther, zfin","panther"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>subject.taxon<\/th>\n      <th>subject.id<\/th>\n      <th>subject<\/th>\n      <th>relation<\/th>\n      <th>object<\/th>\n      <th>object.id<\/th>\n      <th>object.taxon<\/th>\n      <th>evidence<\/th>\n      <th>publications<\/th>\n      <th>provided_by<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="troubleshooting-" class="section level2">
<h2 class="hasAnchor">
<a href="#troubleshooting-" class="anchor"></a>Troubleshooting.</h2>
<p>If we encounter problems, such as an empty result, we can always ask for Monarch’s response to our query.</p>
<div id="response-from-monarch-initiative-" class="section level3">
<h3 class="hasAnchor">
<a href="#response-from-monarch-initiative-" class="anchor"></a>Response from Monarch Initiative.</h3>
<p>We returned the response as a monarch_api (S3) class with its own print method</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(results<span class="op">$</span>response)</code></pre></div>
<pre><code>## [1] "monarch_api"</code></pre>
<p>The print method only shows the URL and highest 2 levels of the contents.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results<span class="op">$</span>response</code></pre></div>
<pre><code>## &lt;monarch https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene%3A8314/homologs/?rows=100&amp;fetch_objects=true&amp;format=json&gt;
## &lt;Showing parsed R objects from json response. 2 level(s) deep&gt;
## List of 7
##  $ compact_associations: NULL
##  $ objects             : chr [1:17] "FlyBase:FBgn0262166" "HGNC:12513" "HGNC:12515" "HGNC:19678" ...
##  $ associations        :'data.frame':    17 obs. of  14 variables:
##   ..$ type              : logi [1:17] NA NA NA NA NA NA ...
##   ..$ qualifiers        : logi [1:17] NA NA NA NA NA NA ...
##   ..$ evidence_graph    :'data.frame':   17 obs. of  2 variables:
##   ..$ evidence_types    : logi [1:17] NA NA NA NA NA NA ...
##   ..$ publications      :List of 17
##   ..$ relation          :'data.frame':   17 obs. of  2 variables:
##   ..$ id                : chr [1:17] "beb2860f-b6ab-472e-8c77-eb8ff2a5da99" "47931d07-86d6-40a2-82d8-d7919a0e8ad5" "5eea0259-b923-49f5-a344-86820defa027" "bfe3507a-f8bf-4a7d-949b-ab3e63dd8e95" ...
##   ..$ subject_extensions: logi [1:17] NA NA NA NA NA NA ...
##   ..$ negated           : logi [1:17] FALSE FALSE FALSE FALSE FALSE FALSE ...
##   ..$ slim              : logi [1:17] NA NA NA NA NA NA ...
##   ..$ subject           :'data.frame':   17 obs. of  3 variables:
##   ..$ object_extensions : logi [1:17] NA NA NA NA NA NA ...
##   ..$ provided_by       :List of 17
##   ..$ object            :'data.frame':   17 obs. of  3 variables:
##  $ start               : NULL
##  $ numFound            : NULL
##  $ facet_pivot         : NULL
##  $ facet_counts        :List of 2
##   ..$ subject_taxon_label:List of 1
##   ..$ object_closure     :List of 25</code></pre>
<p>To view at an alternative depth, supply the <em>depth</em> setting to the print method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># print(results$response, depth=3) # not run.</span></code></pre></div>
<div id="troubleshooting-the-server-status" class="section level4">
<h4 class="hasAnchor">
<a href="#troubleshooting-the-server-status" class="anchor"></a>Troubleshooting the server status</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(results<span class="op">$</span>response<span class="op">$</span>response) <span class="co"># this is the response we saved from the call to httr GET(url)</span></code></pre></div>
<pre><code>## [1] "response"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results<span class="op">$</span>response<span class="op">$</span>url</code></pre></div>
<pre><code>## [1] "https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene%3A8314/homologs/?rows=100&amp;fetch_objects=true&amp;format=json"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results<span class="op">$</span>response<span class="op">$</span>response<span class="op">$</span>status_code</code></pre></div>
<pre><code>## [1] 200</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#str(results$response$response)</span>
<span class="co">#results$response$response$headers</span>
<span class="co">#results$response$response$request</span></code></pre></div>
<p>The server should have responded with an OK (200) status. If not, we can confirm the URL ourselves by pasting into a browser, or look at the response and response headers for more info (courtesy of httr).</p>
</div>
</div>
<div id="view-json-directly-from-the-server-" class="section level3">
<h3 class="hasAnchor">
<a href="#view-json-directly-from-the-server-" class="anchor"></a>View JSON directly from the server.</h3>
<p>We can use the URL at the top of monarchr’s summary of the response to see the JSON results directly from the server.</p>
<p>(this is easier to view in Firefox):</p>
<p>[<a href="https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene%3A8314/homologs/?rows=100&amp;fetch_objects=true&amp;format=json">https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene%3A8314/homologs/?rows=100&amp;fetch_objects=true&amp;format=json</a>]</p>
</div>
<div id="view-html-on-server-" class="section level3">
<h3 class="hasAnchor">
<a href="#view-html-on-server-" class="anchor"></a>View HTML on server.</h3>
<p>If you prefer to see the expected results as HTML, note how the example of looking up NCBIGene:8314 on Monarch Initiative website ended up at a table for Monarch’s preferred ID (HGNC:950).</p>
<p>[<a href="https://monarchinitiative.org/gene/HGNC:950#homologs" class="uri">https://monarchinitiative.org/gene/HGNC:950#homologs</a>]</p>
<p>Replacing HGNC:950 with NCBIGene ID, we are successfully directed to the same view.</p>
<p>[<a href="https://monarchinitiative.org/gene/NCBIGene:8314#homologs" class="uri">https://monarchinitiative.org/gene/NCBIGene:8314#homologs</a>]</p>
</div>
</div>
<div id="verify-your-search-returns-something-consistent-with-searching-monarch-initiatives-website-directly-" class="section level2">
<h2 class="hasAnchor">
<a href="#verify-your-search-returns-something-consistent-with-searching-monarch-initiatives-website-directly-" class="anchor"></a>Verify your search returns something consistent with searching Monarch Initiative’s website directly.</h2>
<ol style="list-style-type: decimal">
<li>Go to Monarch’s front page: [<a href="https://monarchinitiative.org/" class="uri">https://monarchinitiative.org/</a>]</li>
<li>In the search box, enter and search for: NCBIGene:8314.</li>
<li>Select the exact match (of thousands) from Homo sapiens. (This is the first match and shows the gene name is BAP1.)</li>
<li>Select homologs from the tabs at the top of the next page.</li>
</ol>
<p>You should see something similar to the following view.</p>
<div class="figure">
<img src="readme_images/BAP1_homologs.png" alt="Homologs as seen on Monarch Inititiative Website."><p class="caption">Homologs as seen on Monarch Inititiative Website.</p>
</div>
<p>These results shoud be consistent with what we saw in our tibble above, as displayed by DataTables (R DT package).</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="NA">NA</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>CC0</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Charles Champney Carey <br><small class="roles"> Maintainer, author </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Charles Champney Carey.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
