<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>disease_to_model_organism • monarchr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="disease_to_model_organism">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">monarchr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/disease_to_model_organism.html">disease_to_model_organism</a>
    </li>
    <li>
      <a href="../articles/monarchr-simple-ex.html">Simple Example</a>
    </li>
    <li>
      <a href="../articles/proposal.html">`monarchr` - An R package for Monarch</a>
    </li>
    <li>
      <a href="../articles/troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/charlieccarey/monarchr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>disease_to_model_organism</h1>
                        <h4 class="author">charles carey</h4>
            
            <h4 class="date">02 May, 2018</h4>
      
      <small>Source: <a href="https://github.com/charlieccarey/monarchr/blob/master/vignettes/disease_to_model_organism.Rmd"><code>vignettes/disease_to_model_organism.Rmd</code></a></small>
      <p class="hidden name"><code>disease_to_model_organism.Rmd</code></p>

    </div>

    
    
<div id="goal-generate-a-network-diagram-view-of-cancer-gene-homologs-in-a-model-organism-" class="section level2">
<h2 class="hasAnchor">
<a href="#goal-generate-a-network-diagram-view-of-cancer-gene-homologs-in-a-model-organism-" class="anchor"></a>Goal: Generate a network diagram view of cancer gene homologs in a model organism.</h2>
<p>We’ll look for genes in model organisms that might make useful models for a human cancer phenotype. We’ll display these genes in a graph diagram.</p>
<p>To do this we need to define:</p>
<ol style="list-style-type: decimal">
<li>A cancer phenotype.</li>
<li>Find genes assoiated with that cancer phenotype.
<ul>
<li>possibly ID map those genes.</li>
<li>possibly subset those genes by certain attributes.</li>
</ul>
</li>
<li>Find homologs to those genes in other organisms.</li>
<li>Find interactions to those genes in human and in other organisms.</li>
<li>Display the graph of gene interactions in those other organisms.</li>
<li>Elaborate on 5. by looking for coexpression in tissues relevant to this cancer.</li>
</ol>
<p>We’ll try to do these using monarch as follows:</p>
<ol style="list-style-type: decimal">
<li>We’ll attempt to use Monarch to find a list of cancer phenotypes, and select one of these as our goal.
<ul>
<li>We’ll compare that to using NIH’s The Cancer Genome Atlas <a href="https://cancergenome.nih.gov/">TCGA</a>.</li>
</ul>
</li>
<li>We’ll use Monarch to find what human genes are associated with that cancer phenotype.</li>
<li>We’ll use Monarch to find homologs to those genes in other model organisms.</li>
<li>We’ll use Monarch to find interactions to those genes in the other model organisms (these will define a core network..
<ul>
<li>We’ll use Monarch to find interactions of these interactors (so that we can have some chance of defining a closed network.</li>
</ul>
</li>
<li>We’ll display a view of the networks by organism.</li>
<li>We’ll paint or subset the views according to attributes, possibly using monarch.</li>
</ol>
</div>
<div id="selecting-a-cancer-to-work-with-uveal-melanoma" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-a-cancer-to-work-with-uveal-melanoma" class="anchor"></a>Selecting a cancer to work with: Uveal Melanoma</h2>
<p>To ensure a richly documented dataset, we have selected a cancer from <a href="https://cancergenome.nih.gov/cancersselected">selected cancer listings at TCGA</a> for which both sample collection is completed and public data available.</p>
<p>Under ‘head and neck’ cancer, we found <a href="https://cancergenome.nih.gov/cancersselected/UvealMelanoma">uveal melanoma</a>. We’ll use this for our example.</p>
<p>Before proceeding further, we wondered: “Are there any animal models of uveal melanoma?”</p>
<p>A quick pubmed search did reveal <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=uveal+melanoma+mouse">mouse models for uveal melanoma</a>.</p>
</div>
<div id="uveal-melanoma-as-a-disease-at-monarch-initiative-" class="section level2">
<h2 class="hasAnchor">
<a href="#uveal-melanoma-as-a-disease-at-monarch-initiative-" class="anchor"></a>Uveal Melanoma as a disease at Monarch Initiative.</h2>
<p>It would be great if we could simply search for the disease by name and retrieve the top result.</p>
<div id="using-monarchr-find-a-useful-id-for-uveal-melanoma-" class="section level3">
<h3 class="hasAnchor">
<a href="#using-monarchr-find-a-useful-id-for-uveal-melanoma-" class="anchor"></a>Using monarchr: Find a useful ID for ‘uveal melanoma’.</h3>
<p>Load monarchr.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(monarchr)</code></pre></div>
<p>Search for “uveal melanoma” and display top results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">term &lt;-<span class="st"> "uveal melanoma"</span>
results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/biolink_search.html">biolink_search</a></span>(term)
um &lt;-<span class="st"> </span>results<span class="op">$</span>search_results <span class="co"># keep just the results, dropping the 'response' object.</span>
<span class="co">#um</span>
<span class="kw">head</span>(um)</code></pre></div>
<pre><code>## # A tibble: 6 x 13
##   definition       category label prefix equivalent_iri  id    score edges
##   &lt;chr&gt;            &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;           &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;
## 1 A melanoma deri… disease  uvea… MONDO  http://purl.ob… MOND… 116.… 355  
## 2 A melanoma aris… disease  mixe… MONDO  http://purl.ob… MOND… 105.… 88   
## 3 &lt;NA&gt;             disease  post… MONDO  http://purl.ob… MOND… 105.… 15   
## 4 A uveal melanom… disease  necr… MONDO  http://purl.ob… MOND… 105.… 46   
## 5 A uveal melanom… disease  epit… MONDO  http://purl.ob… MOND… 105.… 77   
## 6 &lt;NA&gt;             disease  susc… MONDO  http://purl.ob… MOND… 105.… 105  
## # ... with 5 more variables: synonym &lt;chr&gt;, `_version_` &lt;chr&gt;,
## #   equivalent_curie &lt;chr&gt;, leaf &lt;chr&gt;, iri &lt;chr&gt;</code></pre>
<p>Note that the results are presented in order of the ‘score’ column. This is the same ordering as we observe doing searches at the website.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># subset(um, select = c(id, score, label, category, synonym))</span>
<span class="kw">head</span>(<span class="kw">subset</span>(um, <span class="dt">select =</span> <span class="kw">c</span>(id, score, label, category, synonym)), <span class="dv">10</span>)</code></pre></div>
<pre><code>## # A tibble: 10 x 5
##    id            score     label      category synonym                    
##    &lt;chr&gt;         &lt;chr&gt;     &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;                      
##  1 MONDO:0006486 116.12919 uveal mel… disease  uvea melanoma (disease), M…
##  2 MONDO:0003910 105.42569 mixed cel… disease  mixed cell uveal melanoma,…
##  3 MONDO:0003927 105.42569 posterior… disease  medium/large size posterio…
##  4 MONDO:0004365 105.42569 necrotic … disease  &lt;NA&gt;                       
##  5 MONDO:0006200 105.42569 epithelio… disease  uvea epithelioid cell mela…
##  6 MONDO:0007966 105.42569 susceptib… disease  MELANOMA, UVEAL            
##  7 MONDO:0004062 97.76348  intermedi… disease  intraocular mixed cell typ…
##  8 HP:0007716    66.65321  Intraocul… Phenoty… Uveal melanoma             
##  9 MONDO:0002661 62.40225  uveal dis… disease  uvea disease, disease of u…
## 10 MP:0010276    58.491478 increased… Phenoty… increased uveal melanoma i…</code></pre>
<p>In this case, the top result is a perfect match for our phrase.</p>
<p>We save the ‘id’ for this top result. The id is the Monarch Disease Onotology (MONDO) ID and will be most useful for pulling other results from monarchr.</p>
<p>We print the definition for ‘uveal melanoma’.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">um_id &lt;-<span class="st"> </span>um[<span class="dv">1</span>,]<span class="op">$</span>id
um[um<span class="op">$</span>id <span class="op">==</span><span class="st"> </span>um_id,]<span class="op">$</span>definition</code></pre></div>
<pre><code>## [1] "A melanoma derived from melanocytes of the uveal tract. It is the most common primary intraocular tumor in the United States and Western Europe. Similar to melanoma of the skin, it is rare in Africa and Asia. Diagnostic procedures include ophthalmoscopic exam, fluorescein angiography and ultrasound. Treatment includes: surgical excision of the eye, iridocyclectomy and tumor resection. Recent treatments also include radiotherapy or photo coagulation. Classification of uveal melanomas recognizes four cell types within these tumors: epithelioid, intermediate, mixed cell, and spindle cell types. The spindle cell type uveal melanomas are further sub-classified as spindle cell type A and spindle cell type B."</code></pre>
<p>This definition is the same text as given in the ‘Overview’ tab from our search of the website in the previous section, <a href="https://monarchinitiative.org/disease/MONDO:0006486">MONDO:0006486</a>.</p>
</div>
</div>
<div id="human-genes-associated-with-uveal-melanoma-" class="section level2">
<h2 class="hasAnchor">
<a href="#human-genes-associated-with-uveal-melanoma-" class="anchor"></a>Human Genes associated with uveal melanoma.</h2>
<p>Fetching the human genes associated with the disease is simple.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">um_genes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioentity_genes_assoc_w_disease.html">bioentity_genes_assoc_w_disease</a></span>(um_id)
um_genes &lt;-<span class="st"> </span>um_genes<span class="op">$</span>genes <span class="co"># ignore the response object.</span>
<span class="co">#um_genes</span>
<span class="kw">head</span>(um_genes)</code></pre></div>
<pre><code>## # A tibble: 6 x 10
##   subject.taxon subject.id subject relation  object object.id object.taxon
##   &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;lgl&gt;       
## 1 Homo sapiens  HGNC:18274 CYSLTR2 is marke… uveal… MONDO:00… NA          
## 2 Homo sapiens  HGNC:24308 CLPTM1L contribu… uveal… MONDO:00… NA          
## 3 Homo sapiens  HGNC:950   BAP1    has phen… uveal… MONDO:00… NA          
## 4 Homo sapiens  HGNC:3250  EIF1AX  is marke… uveal… MONDO:00… NA          
## 5 Homo sapiens  HGNC:4390  GNAQ    has phen… uveal… MONDO:00… NA          
## 6 Homo sapiens  HGNC:9059  PLCB4   is marke… uveal… MONDO:00… NA          
## # ... with 3 more variables: evidence &lt;fct&gt;, publications &lt;fct&gt;,
## #   provided_by &lt;fct&gt;</code></pre>
<p>The HGNC IDs are most useful for further searches with monarchr. Obviously we can subset for the particular relation we want. For this vignette, we take all relation types.</p>
<p>HGNC derives from Human identifiers by the HUGO gene nomenclaure committee <a href="https://www.genenames.org/">HUGO</a>.</p>
<p>Note, the same gene can be returned more than once if there is more than one relation(?) or MONDO id (see notes).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">um_hgnc &lt;-<span class="st"> </span>um_genes<span class="op">$</span>subject.id
dups &lt;-<span class="st"> </span>um_genes[<span class="kw">duplicated</span>(um_genes<span class="op">$</span>subject.id),]<span class="op">$</span>subject.id
um_genes[um_genes<span class="op">$</span>subject.id <span class="op">%in%</span><span class="st"> </span>dups, ]</code></pre></div>
<pre><code>## # A tibble: 4 x 10
##   subject.taxon subject.id subject relation  object object.id object.taxon
##   &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;lgl&gt;       
## 1 Homo sapiens  HGNC:4390  GNAQ    has phen… uveal… MONDO:00… NA          
## 2 Homo sapiens  HGNC:4390  GNAQ    likely_p… susce… MONDO:00… NA          
## 3 Homo sapiens  HGNC:4379  GNA11   likely_p… susce… MONDO:00… NA          
## 4 Homo sapiens  HGNC:4379  GNA11   is marke… uveal… MONDO:00… NA          
## # ... with 3 more variables: evidence &lt;fct&gt;, publications &lt;fct&gt;,
## #   provided_by &lt;fct&gt;</code></pre>
<p>Save the unique gene IDs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">um_genes &lt;-<span class="st"> </span><span class="kw">unique</span>(um_genes<span class="op">$</span>subject.id)</code></pre></div>
</div>
<div id="homologs-of-human-genes-" class="section level2">
<h2 class="hasAnchor">
<a href="#homologs-of-human-genes-" class="anchor"></a>Homologs of human genes.</h2>
<p>Find homologs to each of the human genes.</p>
<p>We’d like to use something like the bioentityset function ‘homologs’ to do this all at once. But it doesn’t seem to be working. I tried with our HGNC genes as well as with the NCBI example genes in their supplied example. We’ll watch as the API evolves. Until then, we have to search for each gene’s homolog one at a time.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">homologs &lt;-<span class="st"> </span><span class="kw">lapply</span>(um_genes, bioentity_homologs)
homologs &lt;-<span class="st"> </span><span class="kw">lapply</span>(homologs, <span class="st">"[["</span>, <span class="st">"homologs"</span>)
homologs &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">'rbind'</span>, homologs)</code></pre></div>
</div>
<div id="subset-homologs-to-some-organisms-" class="section level2">
<h2 class="hasAnchor">
<a href="#subset-homologs-to-some-organisms-" class="anchor"></a>Subset homologs to some organisms.</h2>
<p>Subset to some of the more completely researched model organisms. Note that most or all of the mouse and rat are in 1 to 1 orthology relationships, and the other organisms a little less so.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#unique(homologs$object.taxon) # all we can choose from</span>
orgs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">'Mus musculus'</span>, <span class="st">"Rattus norvegicus"</span>, <span class="st">"Danio rerio"</span>, <span class="st">"Drosophila melanogaster"</span>, <span class="st">"Caenorhabditis elegans"</span>, <span class="st">"Saccharomyces cerevisiae S288C"</span>)
ho &lt;-<span class="st"> </span>homologs[homologs<span class="op">$</span>object.taxon <span class="op">%in%</span><span class="st"> </span>orgs,]
ho &lt;-<span class="st"> </span><span class="kw">subset</span>(ho, <span class="dt">select=</span> <span class="kw">c</span>(subject.taxon, subject.id, object.id, object.taxon))
ho &lt;-<span class="st"> </span>ho[<span class="kw">order</span>(ho<span class="op">$</span>object.taxon),]
<span class="kw">split</span>(ho, ho<span class="op">$</span>object.taxon) <span class="co"># list these by organism.</span></code></pre></div>
<pre><code>## $`Caenorhabditis elegans`
## # A tibble: 11 x 4
##    subject.taxon subject.id object.id               object.taxon          
##    &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;                   &lt;chr&gt;                 
##  1 Homo sapiens  HGNC:18274 WormBase:WBGene00015000 Caenorhabditis elegans
##  2 Homo sapiens  HGNC:18274 WormBase:WBGene00022584 Caenorhabditis elegans
##  3 Homo sapiens  HGNC:18274 WormBase:WBGene00018344 Caenorhabditis elegans
##  4 Homo sapiens  HGNC:18274 WormBase:WBGene00021328 Caenorhabditis elegans
##  5 Homo sapiens  HGNC:18274 WormBase:WBGene00012275 Caenorhabditis elegans
##  6 Homo sapiens  HGNC:24308 WormBase:WBGene00011761 Caenorhabditis elegans
##  7 Homo sapiens  HGNC:3250  WormBase:WBGene00019162 Caenorhabditis elegans
##  8 Homo sapiens  HGNC:4390  WormBase:WBGene00001196 Caenorhabditis elegans
##  9 Homo sapiens  HGNC:9059  WormBase:WBGene00001177 Caenorhabditis elegans
## 10 Homo sapiens  HGNC:9059  WormBase:WBGene00004037 Caenorhabditis elegans
## 11 Homo sapiens  HGNC:10768 WormBase:WBGene00011605 Caenorhabditis elegans
## 
## $`Danio rerio`
## # A tibble: 13 x 4
##    subject.taxon subject.id object.id                 object.taxon
##    &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;                     &lt;chr&gt;       
##  1 Homo sapiens  HGNC:18274 ZFIN:ZDB-GENE-071016-2    Danio rerio 
##  2 Homo sapiens  HGNC:18274 ZFIN:ZDB-GENE-071016-1    Danio rerio 
##  3 Homo sapiens  HGNC:24308 ZFIN:ZDB-GENE-040718-75   Danio rerio 
##  4 Homo sapiens  HGNC:950   ZFIN:ZDB-GENE-050208-492  Danio rerio 
##  5 Homo sapiens  HGNC:3250  ZFIN:ZDB-GENE-030131-1319 Danio rerio 
##  6 Homo sapiens  HGNC:3250  ZFIN:ZDB-GENE-041010-185  Danio rerio 
##  7 Homo sapiens  HGNC:4390  ZFIN:ZDB-GENE-081104-25   Danio rerio 
##  8 Homo sapiens  HGNC:9059  NCBIGene:100537538        Danio rerio 
##  9 Homo sapiens  HGNC:9059  ZFIN:ZDB-GENE-060627-1    Danio rerio 
## 10 Homo sapiens  HGNC:4379  ZFIN:ZDB-GENE-050208-597  Danio rerio 
## 11 Homo sapiens  HGNC:4379  ZFIN:ZDB-GENE-041121-9    Danio rerio 
## 12 Homo sapiens  HGNC:8101  ZFIN:ZDB-GENE-070718-4    Danio rerio 
## 13 Homo sapiens  HGNC:10768 ZFIN:ZDB-GENE-040827-3    Danio rerio 
## 
## $`Drosophila melanogaster`
## # A tibble: 14 x 4
##    subject.taxon subject.id object.id           object.taxon           
##    &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;               &lt;chr&gt;                  
##  1 Homo sapiens  HGNC:18274 FlyBase:FBgn0039595 Drosophila melanogaster
##  2 Homo sapiens  HGNC:24308 FlyBase:FBgn0030456 Drosophila melanogaster
##  3 Homo sapiens  HGNC:950   FlyBase:FBgn0262166 Drosophila melanogaster
##  4 Homo sapiens  HGNC:3250  FlyBase:FBgn0026250 Drosophila melanogaster
##  5 Homo sapiens  HGNC:4390  FlyBase:FBgn0050054 Drosophila melanogaster
##  6 Homo sapiens  HGNC:4390  FlyBase:FBgn0033756 Drosophila melanogaster
##  7 Homo sapiens  HGNC:4390  FlyBase:FBgn0004435 Drosophila melanogaster
##  8 Homo sapiens  HGNC:9059  FlyBase:FBgn0262738 Drosophila melanogaster
##  9 Homo sapiens  HGNC:8101  FlyBase:FBgn0041150 Drosophila melanogaster
## 10 Homo sapiens  HGNC:8101  FlyBase:FBgn0036329 Drosophila melanogaster
## 11 Homo sapiens  HGNC:8101  FlyBase:FBgn0051693 Drosophila melanogaster
## 12 Homo sapiens  HGNC:8101  FlyBase:FBgn0035332 Drosophila melanogaster
## 13 Homo sapiens  HGNC:8101  FlyBase:FBgn0031649 Drosophila melanogaster
## 14 Homo sapiens  HGNC:10768 FlyBase:FBgn0031266 Drosophila melanogaster
## 
## $`Mus musculus`
## # A tibble: 9 x 4
##   subject.taxon subject.id object.id   object.taxon
##   &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;       
## 1 Homo sapiens  HGNC:18274 MGI:1917336 Mus musculus
## 2 Homo sapiens  HGNC:24308 MGI:2442892 Mus musculus
## 3 Homo sapiens  HGNC:950   MGI:1206586 Mus musculus
## 4 Homo sapiens  HGNC:3250  MGI:1913485 Mus musculus
## 5 Homo sapiens  HGNC:4390  MGI:95776   Mus musculus
## 6 Homo sapiens  HGNC:9059  MGI:107464  Mus musculus
## 7 Homo sapiens  HGNC:4379  MGI:95766   Mus musculus
## 8 Homo sapiens  HGNC:8101  MGI:97454   Mus musculus
## 9 Homo sapiens  HGNC:10768 MGI:1932339 Mus musculus
## 
## $`Rattus norvegicus`
## # A tibble: 9 x 4
##   subject.taxon subject.id object.id   object.taxon     
##   &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;            
## 1 Homo sapiens  HGNC:18274 RGD:619797  Rattus norvegicus
## 2 Homo sapiens  HGNC:24308 RGD:1307896 Rattus norvegicus
## 3 Homo sapiens  HGNC:950   RGD:1311938 Rattus norvegicus
## 4 Homo sapiens  HGNC:3250  RGD:1560198 Rattus norvegicus
## 5 Homo sapiens  HGNC:4390  RGD:620770  Rattus norvegicus
## 6 Homo sapiens  HGNC:9059  RGD:3345    Rattus norvegicus
## 7 Homo sapiens  HGNC:4379  RGD:619749  Rattus norvegicus
## 8 Homo sapiens  HGNC:8101  RGD:2318412 Rattus norvegicus
## 9 Homo sapiens  HGNC:10768 RGD:620148  Rattus norvegicus
## 
## $`Saccharomyces cerevisiae S288C`
## # A tibble: 2 x 4
##   subject.taxon subject.id object.id      object.taxon                  
##   &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;                         
## 1 Homo sapiens  HGNC:3250  SGD:S000004873 Saccharomyces cerevisiae S288C
## 2 Homo sapiens  HGNC:10768 SGD:S000004901 Saccharomyces cerevisiae S288C</code></pre>
</div>
<div id="interactions-to-those-homologs-" class="section level2">
<h2 class="hasAnchor">
<a href="#interactions-to-those-homologs-" class="anchor"></a>Interactions to those homologs.</h2>
<p>TODO(soon):</p>
</div>
<div id="plots-of-the-homologs-and-their-interactions-" class="section level2">
<h2 class="hasAnchor">
<a href="#plots-of-the-homologs-and-their-interactions-" class="anchor"></a>plots of the homologs and their interactions.</h2>
<p>TODO(soon):</p>
<div id="notes" class="section level3">
<h3 class="hasAnchor">
<a href="#notes" class="anchor"></a>Notes:</h3>
<div id="the-website-equivalent-of-searching-for-uveal-melanoma-" class="section level4">
<h4 class="hasAnchor">
<a href="#the-website-equivalent-of-searching-for-uveal-melanoma-" class="anchor"></a>The website equivalent of searching for ‘uveal melanoma’.</h4>
<p>If we were using the Monarch Initiative Website Directly, we would:</p>
<ul>
<li>search <a href="https://monarchinitiative.org/">Monarch Initiative</a> for ‘uveal melanoma’ by entering the term in the search box and clicking ‘Go’</li>
<li>take the first disease shown in the results, it is named ‘uveal melanoma’.</li>
<li>Clicking it, we get to an overview page and other tabs for ‘uveal melanoma’ for the <a href="https://monarchinitiative.org/disease/MONDO:0006486">MONDO ID 0006486</a>
</li>
</ul>
<p>Note the following tabs and counts at the website:</p>
<ul>
<li>overview</li>
<li>Phenotypes (17)</li>
<li>Genes (9)</li>
<li>Models (6)</li>
<li>Variants (9)</li>
<li>Pathways (3)</li>
<li>Compare</li>
</ul>
<p>We’ll be interested in the overview, to make sure we found the disease we are targeting. And we’ll want the genes to move onto additional steps.</p>
<p>In addition, we see that uveal melanoma is classified under ‘melanoma’ which is classified under ‘cancer’. We make note of these MONDO IDs for validation or analysis steps in this or other vignettes.</p>
<ul>
<li>cancer, MONDO:0004992</li>
<li>melanoma, MONDO:0005105</li>
<li>uveal melanoma, MONDO:0006486</li>
</ul>
<p>Note, the entire MONDO ontology can be found here: <a href="http://www.obofoundry.org/ontology/mondo.html">MONDO</a>.</p>
<p>Note, when we searched using monarchr the (label, category, synonym) columns in the tibble correspond to (Term, Category, Matching String) columns as observed at the website <a href="https://monarchinitiative.org/search/uveal%20melanoma">result</a>. The exception is that the website displayed only one of possibly many synonyms until we drilled further into the results. (TODO(?): How is the particular synonym chosen by monarch? It is not obvious.).</p>
<p>Note, the number of unique genes (9) for uveal melanoma using monarchr is the same as listed under the Genes tab at the website.</p>
</div>
<div id="monarch-returns-all-genes-at-and-below-a-disease-ontologys-level-i-e-node-or-leaf-" class="section level4">
<h4 class="hasAnchor">
<a href="#monarch-returns-all-genes-at-and-below-a-disease-ontologys-level-i-e-node-or-leaf-" class="anchor"></a>Monarch returns all genes at and below a disease ontology’s level (i.e. node or leaf).</h4>
<p>When we searched for MONDO:0006486, we also returned genes for MONDO:0007966. Searching monarch’s website for <a href="https://monarchinitiative.org/disease/MONDO:0007966">MONDO:0007966</a> we see it is directly below the MONDO:0006486 ‘uveal melanoma’ term. So this makes sense, monarchr is returning both the genes at the disease ontology level we requested, and all those underneath it as well. You can confirm this by clicking on each of the nodes and leafs at the website below <a href="https://monarchinitiative.org/disease/MONDO:0006486">uveal melonoma</a>. The only leaf that shows genes is ‘susceptibility to melanoma’, all other genes are classified solely at the higher level of ‘uveal melanoma’. (That isn’t the entire story, we do see some variants and phenotypes at the other nodes and leaves.)</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#goal-generate-a-network-diagram-view-of-cancer-gene-homologs-in-a-model-organism-">Goal: Generate a network diagram view of cancer gene homologs in a model organism.</a></li>
      <li><a href="#selecting-a-cancer-to-work-with-uveal-melanoma">Selecting a cancer to work with: Uveal Melanoma</a></li>
      <li><a href="#uveal-melanoma-as-a-disease-at-monarch-initiative-">Uveal Melanoma as a disease at Monarch Initiative.</a></li>
      <li><a href="#human-genes-associated-with-uveal-melanoma-">Human Genes associated with uveal melanoma.</a></li>
      <li><a href="#homologs-of-human-genes-">Homologs of human genes.</a></li>
      <li><a href="#subset-homologs-to-some-organisms-">Subset homologs to some organisms.</a></li>
      <li><a href="#interactions-to-those-homologs-">Interactions to those homologs.</a></li>
      <li><a href="#plots-of-the-homologs-and-their-interactions-">plots of the homologs and their interactions.</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Charles C Carey, Ted Laderas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
