<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Simple Example â€¢ monarchr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="Simple Example">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">monarchr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/disease_to_model_organism.html">disease_to_model_organism</a>
    </li>
    <li>
      <a href="../articles/monarchr-simple-ex.html">Simple Example</a>
    </li>
    <li>
      <a href="../articles/proposal.html">`monarchr` - An R package for Monarch</a>
    </li>
    <li>
      <a href="../articles/troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/charlieccarey/monarchr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="monarchr-simple-ex_files/htmlwidgets-1.2/htmlwidgets.js"></script><script src="monarchr-simple-ex_files/jquery-1.12.4/jquery.min.js"></script><link href="monarchr-simple-ex_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="monarchr-simple-ex_files/datatables-binding-0.4/datatables.js"></script><link href="monarchr-simple-ex_files/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="monarchr-simple-ex_files/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="monarchr-simple-ex_files/dt-core-1.10.16/js/jquery.dataTables.min.js"></script><link href="monarchr-simple-ex_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="monarchr-simple-ex_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Simple Example</h1>
                        <h4 class="author">Charles Champney Carey</h4>
            
            <h4 class="date">2018-05-02</h4>
      
      <small>Source: <a href="https://github.com/charlieccarey/monarchr/blob/master/vignettes/monarchr-simple-ex.Rmd"><code>vignettes/monarchr-simple-ex.Rmd</code></a></small>
      <p class="hidden name"><code>monarchr-simple-ex.Rmd</code></p>

    </div>

    
    
<p>Replicate the 11 tabs from Monarch Initiative that result from a search on a human gene.</p>
<p>Here are a some of them for now.</p>
<p>We load the monarchr client and the DT libraries.</p>
<p>(DT is the datatables package which we use to display the results.)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(monarchr)
<span class="kw">library</span>(DT)
<span class="kw">library</span>(knitr)
gene &lt;-<span class="st"> "NCBIGene:8314"</span></code></pre></div>
<div id="get-the-homologs-paralogs-orthologs-and-homologs-to-a-gene-" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-homologs-paralogs-orthologs-and-homologs-to-a-gene-" class="anchor"></a>Get the homologs (paralogs, orthologs and homologs) to a gene.</h2>
<p>Homologs of gene, displayed using DataTable.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">homs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioentity_homologs.html">bioentity_homologs</a></span>(gene)
<span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(homs<span class="op">$</span>homologs)</code></pre></div>
<div id="htmlwidget-489287a4626203dfbea8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-489287a4626203dfbea8">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17"],["Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens"],["HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950","HGNC:950"],["BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1","BAP1"],["in paralogy relationship with","in paralogy relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in paralogy relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with","in orthology relationship with","in 1 to 1 orthology relationship with"],["UCHL5","UCHL3","BAP1","Bap1","UCH2","Bap1","BAP1","UCHL1","bap1","BAP1","BAP1","BAP1","BAP1","UCH1","bap1","bap1","calypso"],["HGNC:19678","HGNC:12515","NCBIGene:100051934","RGD:1311938","NCBIGene:842876","MGI:1206586","NCBIGene:100124510","HGNC:12513","Xenbase:XB-GENE-494978","NCBIGene:460426","NCBIGene:100030960","NCBIGene:100154798","NCBIGene:697221","NCBIGene:831492","NCBIGene:100557123","ZFIN:ZDB-GENE-050208-492","FlyBase:FBgn0262166"],["Homo sapiens","Homo sapiens","Equus caballus","Rattus norvegicus","Arabidopsis thaliana","Mus musculus","Bos taurus","Homo sapiens","Xenopus (Silurana) tropicalis","Pan troglodytes","Monodelphis domestica","Sus scrofa","Macaca mulatta","Arabidopsis thaliana","Anolis carolinensis","Danio rerio","Drosophila melanogaster"],["phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","phylogenetic evidence","sequence similarity evidence, phylogenetic evidence, protein BLAST evidence used in manual assertion","phylogenetic evidence"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"ZFIN:ZDB-PUB-030905-1",null],["panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther","panther, zfin","panther"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>subject.taxon<\/th>\n      <th>subject.id<\/th>\n      <th>subject<\/th>\n      <th>relation<\/th>\n      <th>object<\/th>\n      <th>object.id<\/th>\n      <th>object.taxon<\/th>\n      <th>evidence<\/th>\n      <th>publications<\/th>\n      <th>provided_by<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="get-the-diseases-associated-with-a-gene-" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-diseases-associated-with-a-gene-" class="anchor"></a>Get the diseases associated with a gene.</h2>
<p>Diseases associated with gene, displayed using kable.</p>
<p>Note, the publications look clickable, but are not. This is a bug in the formatting. They are meant to be raw text.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dis &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioentity_diseases_assoc_w_gene.html">bioentity_diseases_assoc_w_gene</a></span>(gene)
<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(dis<span class="op">$</span>diseases)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">subject.taxon</th>
<th align="left">subject.id</th>
<th align="left">subject</th>
<th align="left">relation</th>
<th align="left">object</th>
<th align="left">object.id</th>
<th align="left">object.taxon</th>
<th align="left">evidence</th>
<th align="left">publications</th>
<th align="left">provided_by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">lipoma</td>
<td align="left">MONDO:0005106</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">leukemia (disease)</td>
<td align="left">MONDO:0005059</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">bone cancer</td>
<td align="left">MONDO:0002129</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">epithelioid and spindle cell nevus</td>
<td align="left">MONDO:0006749</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">testicular cancer</td>
<td align="left">MONDO:0005447</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">meningioma (disease)</td>
<td align="left">MONDO:0016642</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">has phenotype</td>
<td align="left">uveal melanoma</td>
<td align="left">MONDO:0006486</td>
<td align="left">NA</td>
<td align="left">traceable author statement, imported manually asserted information used in automatic assertion</td>
<td align="left">
<a href="PMID:25231345" class="uri">PMID:25231345</a>, <a href="PMID:26719535" class="uri">PMID:26719535</a>, <a href="PMID:21874000" class="uri">PMID:21874000</a>
</td>
<td align="left">ctd, orphanet</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">malignant mesothelioma (disease)</td>
<td align="left">MONDO:0006292</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left">
<a href="PMID:25231345" class="uri">PMID:25231345</a>, <a href="PMID:26463840" class="uri">PMID:26463840</a>, <a href="PMID:29112861" class="uri">PMID:29112861</a>, <a href="PMID:26928227" class="uri">PMID:26928227</a>, <a href="PMID:24928783" class="uri">PMID:24928783</a>
</td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">neuroendocrine carcinoma</td>
<td align="left">MONDO:0002120</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">lung adenocarcinoma</td>
<td align="left">MONDO:0005061</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">mesothelioma</td>
<td align="left">MONDO:0005065</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left">
<a href="PMID:26119930" class="uri">PMID:26119930</a>, <a href="PMID:26719535" class="uri">PMID:26719535</a>, <a href="PMID:21874000" class="uri">PMID:21874000</a>, <a href="PMID:21642991" class="uri">PMID:21642991</a>
</td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">mucoepidermoid carcinoma</td>
<td align="left">MONDO:0003036</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">central nervous system cancer</td>
<td align="left">MONDO:0002714</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">kidney cancer</td>
<td align="left">MONDO:0002367</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:26719535" class="uri">PMID:26719535</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">colonic neoplasm</td>
<td align="left">MONDO:0005401</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">paraganglioma</td>
<td align="left">MONDO:0000448</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">renal cell carcinoma (disease)</td>
<td align="left">MONDO:0005086</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left">
<a href="PMID:25231345" class="uri">PMID:25231345</a>, <a href="PMID:23797736" class="uri">PMID:23797736</a>, <a href="PMID:22683710" class="uri">PMID:22683710</a>, <a href="PMID:22138691" class="uri">PMID:22138691</a>
</td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">pancreatic neoplasm</td>
<td align="left">MONDO:0021040</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">likely_pathogenic_for_condition</td>
<td align="left">BAP1-related tumor predisposition syndrome</td>
<td align="left">MONDO:0013692</td>
<td align="left">NA</td>
<td align="left">imported manually asserted information used in automatic assertion, sequencing assay evidence</td>
<td align="left">
<a href="PMID:21941004" class="uri">PMID:21941004</a>, <a href="PMID:23684012" class="uri">PMID:23684012</a>, <a href="PMID:21874000" class="uri">PMID:21874000</a>, <a href="PMID:21874003" class="uri">PMID:21874003</a>
</td>
<td align="left">omim, ctd, orphanet, clinvar</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">gastric neoplasm</td>
<td align="left">MONDO:0021085</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left">
<a href="PMID:25231345" class="uri">PMID:25231345</a>, <a href="PMID:26719535" class="uri">PMID:26719535</a>
</td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">contributes to</td>
<td align="left">familial melanoma</td>
<td align="left">MONDO:0018961</td>
<td align="left">NA</td>
<td align="left">imported manually asserted information used in automatic assertion</td>
<td align="left">NA</td>
<td align="left">orphanet</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">cholangiocarcinoma</td>
<td align="left">MONDO:0019087</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left">
<a href="PMID:25231345" class="uri">PMID:25231345</a>, <a href="PMID:24185509" class="uri">PMID:24185509</a>
</td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">ovarian cancer</td>
<td align="left">MONDO:0008170</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">nasopharyngeal carcinoma</td>
<td align="left">MONDO:0015459</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:24952746" class="uri">PMID:24952746</a></td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">is marker for</td>
<td align="left">uterine cancer</td>
<td align="left">MONDO:0002715</td>
<td align="left">NA</td>
<td align="left">traceable author statement</td>
<td align="left"><a href="PMID:25231345" class="uri">PMID:25231345</a></td>
<td align="left">ctd</td>
</tr>
</tbody>
</table>
</div>
<div id="get-the-phenotypes-associated-with-a-gene-" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-phenotypes-associated-with-a-gene-" class="anchor"></a>Get the phenotypes associated with a gene.</h2>
<p>Phenotypes associated with gene, displayed using kable.</p>
<p>Note, the publications look clickable, but are not. This is a bug in the formatting. They are meant to be raw text.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">phe &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioentity_phenotypes_assoc_w_gene.html">bioentity_phenotypes_assoc_w_gene</a></span>(gene)
<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(phe<span class="op">$</span>phenotypes))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">subject.taxon</th>
<th align="left">subject.id</th>
<th align="left">subject</th>
<th align="left">relation</th>
<th align="left">object</th>
<th align="left">object.id</th>
<th align="left">object.taxon</th>
<th align="left">evidence</th>
<th align="left">publications</th>
<th align="left">provided_by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">has phenotype</td>
<td align="left">Metamorphopsia</td>
<td align="left">HP:0012508</td>
<td align="left">NA</td>
<td align="left">traceable author statement used in manual assertion, imported manually asserted information used in automatic assertion</td>
<td align="left">NA</td>
<td align="left">hpoa, orphanet</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">has phenotype</td>
<td align="left">Zonular cataract</td>
<td align="left">HP:0010920</td>
<td align="left">NA</td>
<td align="left">traceable author statement used in manual assertion, imported manually asserted information used in automatic assertion</td>
<td align="left">NA</td>
<td align="left">hpoa, orphanet</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">has phenotype</td>
<td align="left">Intraocular melanoma</td>
<td align="left">HP:0007716</td>
<td align="left">NA</td>
<td align="left">evidence used in automatic assertion, imported manually asserted information used in automatic assertion, sequencing assay evidence</td>
<td align="left">
<a href="PMID:21941004" class="uri">PMID:21941004</a>, <a href="PMID:23684012" class="uri">PMID:23684012</a>, <a href="PMID:21874000" class="uri">PMID:21874000</a>, <a href="PMID:21874003" class="uri">PMID:21874003</a>
</td>
<td align="left">omim, hpoa, orphanet, clinvar</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">has phenotype</td>
<td align="left">Mydriasis</td>
<td align="left">HP:0011499</td>
<td align="left">NA</td>
<td align="left">traceable author statement used in manual assertion, imported manually asserted information used in automatic assertion</td>
<td align="left">NA</td>
<td align="left">hpoa, orphanet</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">has phenotype</td>
<td align="left">Choroidal melanoma</td>
<td align="left">HP:0012054</td>
<td align="left">NA</td>
<td align="left">traceable author statement used in manual assertion, imported manually asserted information used in automatic assertion</td>
<td align="left">NA</td>
<td align="left">hpoa, orphanet</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">has phenotype</td>
<td align="left">Ocular pain</td>
<td align="left">HP:0200026</td>
<td align="left">NA</td>
<td align="left">traceable author statement used in manual assertion, imported manually asserted information used in automatic assertion</td>
<td align="left">NA</td>
<td align="left">hpoa, orphanet</td>
</tr>
</tbody>
</table>
</div>
<div id="get-the-anatomy-associated-with-a-gene-" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-anatomy-associated-with-a-gene-" class="anchor"></a>Get the anatomy associated with a gene.</h2>
<p>Anatomy associated with gene, displayed using kable.</p>
<p>Molecularly defined by expression data?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">anat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioentity_exp_anatomy_assoc_w_gene.html">bioentity_exp_anatomy_assoc_w_gene</a></span>(gene)
<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(anat<span class="op">$</span>anatomy))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">subject.taxon</th>
<th align="left">subject.id</th>
<th align="left">subject</th>
<th align="left">relation</th>
<th align="left">object</th>
<th align="left">object.id</th>
<th align="left">object.taxon</th>
<th align="left">evidence</th>
<th align="left">publications</th>
<th align="left">provided_by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">expressed in</td>
<td align="left">vastus lateralis</td>
<td align="left">UBERON:0001379</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">bgee</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">expressed in</td>
<td align="left">central nervous system</td>
<td align="left">UBERON:0001017</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">bgee</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">expressed in</td>
<td align="left">nucleus accumbens</td>
<td align="left">UBERON:0001882</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">bgee</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">expressed in</td>
<td align="left">layer of synovial tissue</td>
<td align="left">UBERON:0007616</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">bgee</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">expressed in</td>
<td align="left">paraflocculus</td>
<td align="left">UBERON:0005351</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">bgee</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">expressed in</td>
<td align="left">Brodmann (1909) area 46</td>
<td align="left">UBERON:0006483</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">bgee</td>
</tr>
</tbody>
</table>
</div>
<div id="get-the-pathways-associated-with-a-gene-" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-pathways-associated-with-a-gene-" class="anchor"></a>Get the pathways associated with a gene.</h2>
<p>Pathways associated with gene, displayed using kable.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pth &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioentity_pathways_assoc_w_gene.html">bioentity_pathways_assoc_w_gene</a></span>(gene)
<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(pth<span class="op">$</span>pathways)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">subject.taxon</th>
<th align="left">subject.id</th>
<th align="left">subject</th>
<th align="left">relation</th>
<th align="left">object</th>
<th align="left">object.id</th>
<th align="left">object.taxon</th>
<th align="left">evidence</th>
<th align="left">publications</th>
<th align="left">provided_by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">UCH proteinases</td>
<td align="left">REACT:R-HSA-5689603</td>
<td align="left">NA</td>
<td align="left">traceable author statement used in manual assertion</td>
<td align="left">NA</td>
<td align="left">ctd, reactome</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">Post-translational protein modification</td>
<td align="left">REACT:R-HSA-597592</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">Recruitment and ATM-mediated phosphorylation of repair and signaling proteins at DNA double strand breaks</td>
<td align="left">REACT:R-HSA-5693565</td>
<td align="left">NA</td>
<td align="left">traceable author statement used in manual assertion</td>
<td align="left">NA</td>
<td align="left">ctd, reactome</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">Deubiquitination</td>
<td align="left">REACT:R-HSA-5688426</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">Metabolism of proteins</td>
<td align="left">REACT:R-HSA-392499</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">DNA Repair</td>
<td align="left">REACT:R-HSA-73894</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ctd</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">DNA Double Strand Break Response</td>
<td align="left">REACT:R-HSA-5693606</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ctd</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">involved in</td>
<td align="left">DNA Double-Strand Break Repair</td>
<td align="left">REACT:R-HSA-5693532</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ctd</td>
</tr>
</tbody>
</table>
</div>
<div id="get-the-interactions-associated-with-a-gene-" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-interactions-associated-with-a-gene-" class="anchor"></a>Get the interactions associated with a gene.</h2>
<p>Interactions associated with gene, displayed using kable.</p>
<p>Note, the publications look clickable, but are not. This is a bug in the formatting. They are meant to be raw text.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">intx &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioentity_interactions_assoc_w_gene.html">bioentity_interactions_assoc_w_gene</a></span>(gene)
<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(intx<span class="op">$</span>interactions))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">subject.taxon</th>
<th align="left">subject.id</th>
<th align="left">subject</th>
<th align="left">relation</th>
<th align="left">object</th>
<th align="left">object.id</th>
<th align="left">object.taxon</th>
<th align="left">evidence</th>
<th align="left">publications</th>
<th align="left">provided_by</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">interacts with</td>
<td align="left">IPO4</td>
<td align="left">HGNC:19426</td>
<td align="left">Homo sapiens</td>
<td align="left">affinity chromatography evidence</td>
<td align="left"><a href="PMID:19615732" class="uri">PMID:19615732</a></td>
<td align="left">biogrid</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">interacts with</td>
<td align="left">USP28</td>
<td align="left">HGNC:12625</td>
<td align="left">Homo sapiens</td>
<td align="left">enzyme assay evidence</td>
<td align="left"><a href="PMID:22626734" class="uri">PMID:22626734</a></td>
<td align="left">biogrid</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">interacts with</td>
<td align="left">MAP3K1</td>
<td align="left">HGNC:6848</td>
<td align="left">Homo sapiens</td>
<td align="left">enzyme assay evidence</td>
<td align="left"><a href="PMID:25260751" class="uri">PMID:25260751</a></td>
<td align="left">biogrid</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">interacts with</td>
<td align="left">XRCC3</td>
<td align="left">HGNC:12830</td>
<td align="left">Homo sapiens</td>
<td align="left">yeast 2-hybrid evidence</td>
<td align="left"><a href="PMID:25640309" class="uri">PMID:25640309</a></td>
<td align="left">biogrid</td>
</tr>
<tr class="odd">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">interacts with</td>
<td align="left">PSMB2</td>
<td align="left">HGNC:9539</td>
<td align="left">Homo sapiens</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">string</td>
</tr>
<tr class="even">
<td align="left">Homo sapiens</td>
<td align="left">HGNC:950</td>
<td align="left">BAP1</td>
<td align="left">interacts with</td>
<td align="left">AHCYL2</td>
<td align="left">HGNC:22204</td>
<td align="left">Homo sapiens</td>
<td align="left">affinity chromatography evidence</td>
<td align="left"><a href="PMID:19615732" class="uri">PMID:19615732</a></td>
<td align="left">biogrid</td>
</tr>
</tbody>
</table>
<!-- Vignettes are long form documentation commonly included in packages. Because they are part of the distribution of the package, they need to be as compact as possible. The `html_vignette` output type provides a custom style sheet (and tweaks some options) to ensure that the resulting html is as small as possible. The `html_vignette` format: --><!-- - Never uses retina figures --><!-- - Has a smaller default figure size --><!-- - Uses a custom CSS stylesheet instead of the default Twitter Bootstrap style --><!-- ## Vignette Info --><!-- Note the various macros within the `vignette` section of the metadata block above. These are required in order to instruct R how to build the vignette. Note that you should change the `title` field and the `\VignetteIndexEntry` to match the title of your vignette. --><!-- ## Styles --><!-- The `html_vignette` template includes a basic CSS theme. To override this theme you can specify your own CSS in the document metadata as follows: --><!--     output:  --><!--       rmarkdown::html_vignette: --><!--         css: mystyles.css --><!-- ## Figures --><!-- The figure sizes have been customised so that you can easily put two images side-by-side.  --><!-- ```{r, fig.show='hold'} --><!-- plot(1:10) --><!-- plot(10:1) --><!-- ``` --><!-- You can enable figure captions by `fig_caption: yes` in YAML: --><!--     output: --><!--       rmarkdown::html_vignette: --><!--         fig_caption: yes --><!-- Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**. --><!-- ## More Examples --><!-- You can write math expressions, e.g. $Y = X\beta + \epsilon$, footnotes^[A footnote here.], and tables, e.g. using `knitr::kable()`. --><!-- ```{r, echo=FALSE, results='asis'} --><!-- knitr::kable(head(mtcars, 10)) --><!-- ``` --><!-- Also a quote using `>`: --><!-- > "He who gives up [code] safety for [code] speed deserves neither." --><!-- ([via](https://twitter.com/hadleywickham/status/504368538874703872)) -->
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#get-the-homologs-paralogs-orthologs-and-homologs-to-a-gene-">Get the homologs (paralogs, orthologs and homologs) to a gene.</a></li>
      <li><a href="#get-the-diseases-associated-with-a-gene-">Get the diseases associated with a gene.</a></li>
      <li><a href="#get-the-phenotypes-associated-with-a-gene-">Get the phenotypes associated with a gene.</a></li>
      <li><a href="#get-the-anatomy-associated-with-a-gene-">Get the anatomy associated with a gene.</a></li>
      <li><a href="#get-the-pathways-associated-with-a-gene-">Get the pathways associated with a gene.</a></li>
      <li><a href="#get-the-interactions-associated-with-a-gene-">Get the interactions associated with a gene.</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Charles C Carey, Ted Laderas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
