<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>proposal: <code>monarchr</code> - An R package for Monarch • monarchr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="proposal: &lt;code&gt;monarchr&lt;/code&gt; - An R package for Monarch">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">monarchr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/disease_to_model_organism.html">disease to model organism</a>
    </li>
    <li>
      <a href="../articles/proposal.html">proposal: `monarchr` - An R package for Monarch</a>
    </li>
    <li>
      <a href="../articles/simple_ex_gene_info.html">Simple Example: info for a gene</a>
    </li>
    <li>
      <a href="../articles/troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/charlieccarey/monarchr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>proposal: <code>monarchr</code> - An R package for Monarch</h1>
                        <h4 class="author">Charles Carey and Ted Laderas</h4>
            
            <h4 class="date">2018-05-09</h4>
      
      <small>Source: <a href="https://github.com/charlieccarey/monarchr/blob/master/vignettes/proposal.Rmd"><code>vignettes/proposal.Rmd</code></a></small>
      <p class="hidden name"><code>proposal.Rmd</code></p>

    </div>

    
    
<p>R/<a href="https://www.bioconductor.org">Bioconductor</a> – with approximately 1,500 software, 325 experiment data and 900 annotation packages and thousands of users – is highly used in bioinformatics workflows and teaching environments. It is a rich source of both experimental data and annotations, allowing users to quickly connect identifiers that are technology specific to gene/transcript/protein identifiers.</p>
<p>While Bioconductor has many packages for annotation, mapping disease descriptions is a relatively underserved area. An R-based interface to Monarch Ontologies, including the Human Phenotype Ontology (HPO) and the Monarch Disease Ontology (MONDO) would not only increase use of these ontologies, but also enable better open science. Further, we anticipate that the variety of ID and annotation types available in a one-stop manner from Monarch Initiative would be valued by the community.</p>
<p>We are suggesting to develop the <code>monarchr</code> package based on the <a href="https://www.bioconductor.org/help/workflows/annotation-data/">Bioconductor annotation workflow</a>.</p>
<div id="starting-steps-using-swagger-on-the-biolink-api" class="section level2">
<h2 class="hasAnchor">
<a href="#starting-steps-using-swagger-on-the-biolink-api" class="anchor"></a>Starting Steps: Using Swagger on the Biolink API</h2>
<p><a href="https://swagger.io/swagger-codegen/">Swagger Codegen</a> is a code generation framework that is helpful in developing API clients for any API that fits the OpenAPI specification. API clients can be developed, tested and deployed in an automated fashion for a number of languages (including R, Python and Rust). In R, we’ve found a need to further modify the client code. Additionally, we might want to provide additional convenience functions for common tasks that require multiple calls to the API.</p>
<p>Using Swagger Codegen on the Biolink API will be the starting point for the <code>monarchr</code> package, as it generates stubs and auto-documentation in the R Package format. However, putting together the functions into use cases that are useful is the vital next step in development.</p>
</div>
<div id="current-mockup-code-and-simple-vignettes" class="section level2">
<h2 class="hasAnchor">
<a href="#current-mockup-code-and-simple-vignettes" class="anchor"></a>Current Mockup: Code and Simple Vignettes</h2>
<p>A partially functional example of monarchr with some simple vignettes is available, along with this proposal at:</p>
<ul>
<li>
<a href="https://github.com/charlieccarey/monarchr">monarchr on github</a> <a href="https://github.com/charlieccarey/monarchr" class="uri">https://github.com/charlieccarey/monarchr</a>
</li>
<li>
<a href="https://charlieccarey.github.io/monarchr/index.html">documentation and vignettes (as articles)</a> <a href="https://charlieccarey.github.io/monarchr/index.html" class="uri">https://charlieccarey.github.io/monarchr/index.html</a>
</li>
</ul>
</div>
<div id="development-of-monarch-use-cases" class="section level2">
<h2 class="hasAnchor">
<a href="#development-of-monarch-use-cases" class="anchor"></a>Development of Monarch Use Cases</h2>
<p>Swagger Codegen is just the starting point for <code>monarchr</code>. The core of development will be showcasing uses of monarchr. Especially, but not exclusively, we imagine good use cases exploring phenotype and disease ontologies (esp. MONDO and HPO). This development will be <code>vignette</code>/documentation driven, where Monarch developers and users will describe a use case for <code>monarchr</code> (such as mapping diseases to genes and their homologs). We’ll show how to accomplish selected goals in ways that are easily extensible by users.</p>
</div>
<div id="submission-of-monarchr-as-a-bioconductor-annotation-package" class="section level2">
<h2 class="hasAnchor">
<a href="#submission-of-monarchr-as-a-bioconductor-annotation-package" class="anchor"></a>Submission of <code>monarchr</code> as a Bioconductor annotation package</h2>
<p>We are recommending that the package be part of <a href="https://www.bioconductor.org">Bioconductor</a> as it has the largest potential audience of users who would want to leverage MONDO and HPO. Doing this would add needed resource/development time to <code>monarchr</code>.</p>
<p><a href="https://www.bioconductor.org/developers/package-guidelines/">Submission Guidelines to Bioconductor</a> are fairly strict. The data structures output should be current Bioconductor objects (such as <code>ExpressionSet</code> and <code>AnnotatedDataFrame</code>). Bioconductor also has strict instructions on <a href="https://bioconductor.org/developers/how-to/web-query/">web query packages</a> to ensure that the package continues to work or fails gracefully.</p>
<p>Packages must also pass their nightly build tests on a number of machines. Using a continuous integration framework such as <a href="https://travis-ci.org/getting_started">Travis CI</a> will help accomplish this.</p>
<p>Bioconductor is released twice a year. We would target the October 2018 release. Interim releases would be made available on github with easy to follow installation instructions.</p>
</div>
<div id="future-directions" class="section level1">
<h1 class="hasAnchor">
<a href="#future-directions" class="anchor"></a>Future Directions:</h1>
<div id="python-client" class="section level2">
<h2 class="hasAnchor">
<a href="#python-client" class="anchor"></a>Python Client</h2>
<p>Depending on funding, it is possible to use Swagger to generate and showcase an API Client for Python as well. The package community for Python in biology is less organized and less centralized than for Bioconductor. A main set of tools is offered by <a href="http://biopython.org/">biopython</a>. We would demonstrate the client in conjunction with many common biopython tasks.</p>
</div>
<div id="outreach" class="section level2">
<h2 class="hasAnchor">
<a href="#outreach" class="anchor"></a>Outreach</h2>
<p>Depending on funding, the July 2018 Bioconductor conference <a href="http://bioc2018.bioconductor.org/schedule">BioC 2018</a> might be a useful place to introduce the R community to <code>monarchr</code>. Additionally it might be the best place to seek feedback and further elaborate on vignette goals.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#starting-steps-using-swagger-on-the-biolink-api">Starting Steps: Using Swagger on the Biolink API</a></li>
      <li><a href="#current-mockup-code-and-simple-vignettes">Current Mockup: Code and Simple Vignettes</a></li>
      <li><a href="#development-of-monarch-use-cases">Development of Monarch Use Cases</a></li>
      <li><a href="#submission-of-monarchr-as-a-bioconductor-annotation-package">Submission of <code>monarchr</code> as a Bioconductor annotation package</a></li>
      <li>
<a href="#future-directions">Future Directions:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#python-client">Python Client</a></li>
      <li><a href="#outreach">Outreach</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Charles C Carey, Ted Laderas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
